src/NLPModels.jl

```diff
@@ -144,6 +144,20 @@ Evaluate \$\\nabla c(x)^Tv\$, the transposed-Jacobian-vector product at `x` in p
 jtprod!(::AbstractNLPModel, ::AbstractVector, ::AbstractVector, ::AbstractVector) =
   throw(NotImplementedError("jtprod!"))
 
+"""`J = jac_op(nlp, x)`
+
+Return the Jacobian at `x` as a linear operator.
+The resulting object may be used as if it were a matrix, e.g., `J * v` or
+`J' * v`.
+"""
+function jac_op(nlp :: AbstractNLPModel, x :: Vector{Float64})
+  return LinearOperator(nlp.meta.ncon, nlp.meta.nvar,
+                        false, false,
+                        v -> jprod(nlp, x, v),
+                        Nullable{Function}(),
+                        v -> jtprod(nlp, x, v))
+end
+
```

Comment:
It might have to be `LinearOperator{Float64}` because by default, when an operator is neither symmetric not hermitian, it's considered to be Complex128. That's a wart I feel but I don't have a more elegant constructor for now.